src/averageAges.js

```diff
@@ -55,7 +85,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = !onlyWithSon
+    ? people.filter(child =>
+      (people.find(person => child.mother === person.name))
+    )
+    : people.filter(child =>
+      (people.find(person => child.mother === person.name && child.sex === 'm'))
+    );
+
+  const ageDiff = children.map(child => (
+    child.born - people.find(person => (
+      person.name === child.mother)).born));
+
+  return ageDiff.reduce((a, b) => a + b) / ageDiff.length;
```

Comment:
would be totally great to have helper here

Reply:
ty for review and great advices. it`s clear for me now :)